test/SpokePool.Relay.ts

```diff
@@ -325,27 +345,345 @@ describe("SpokePool Relayer Logic", async function () {
+          )
+        ).to.changeTokenBalance(destErc20, spokePool, relayExecution.updatedOutputAmount.mul(-1));
+      });
+      it("if recipient is contract that implements message handler, calls message handler", async function () {
+        // Does nothing if message length is 0
+        const acrossMessageHandler = await createFake("AcrossMessageHandlerMock");
+        const _relayData = {
+          ...relayData,
+          recipient: acrossMessageHandler.address,
+          message: "0x1234",
+        };
+        const relayExecution = await getRelayExecutionParams(_relayData);
+
+        // Handler is called with expected params.
+        await spokePool.connect(relayer).fillRelayUSSInternal(
+          relayExecution,
+          relayer.address,
+          false // isSlowFill
+        );
+        expect(acrossMessageHandler.handleUSSAcrossMessage).to.have.been.calledOnceWith(
+          _relayData.outputToken,
+          relayExecution.updatedOutputAmount,
+          relayer.address, // Custom relayer
+          _relayData.message
+        );
+      });
+    });
+    describe("fillUSSRelay", function () {
+      it("fills are not paused", async function () {
+        await spokePool.pauseFills(true);
+        await expect(spokePool.connect(relayer).fillUSSRelay(relayData, consts.repaymentChainId)).to.be.revertedWith(
+          "Paused fills"
+        );
+      });
+      it("reentrancy protected", async function () {
+        const callbackMessageHandler = await (
+          await getContractFactory("AcrossMessageHandlerCallbackMock", depositor)
+        ).deploy();
+        const functionCalldata = spokePool.interface.encodeFunctionData("fillUSSRelay", [
+          relayData,
+          consts.repaymentChainId,
+        ]);
+        const _relayData = {
+          ...relayData,
+          recipient: callbackMessageHandler.address,
+          message: functionCalldata,
+        };
+        await expect(spokePool.connect(relayer).fillUSSRelay(_relayData, consts.repaymentChainId)).to.be.revertedWith(
+          "ReentrancyGuard: reentrant call"
+        );
+      });
+      it("must be exclusive relayer before exclusivity deadline", async function () {
+        const _relayData = {
+          ...relayData,
+          // Overwrite exclusive relayer and exclusivity deadline
+          exclusiveRelayer: recipient.address,
+          exclusivityDeadline: relayData.fillDeadline,
+        };
+        await expect(spokePool.connect(relayer).fillUSSRelay(_relayData, consts.repaymentChainId)).to.be.revertedWith(
+          "NotExclusiveRelayer"
+        );
+
+        // Can send it after exclusivity deadline
+        await expect(
+          spokePool.connect(relayer).fillUSSRelay(
+            {
+              ..._relayData,
+              exclusivityDeadline: 0,
+            },
+            consts.repaymentChainId
+          )
+        ).to.not.be.reverted;
+      });
+      it("can't overwrite RelayData destination chain", async function () {
```

Comment:
I missed the review where this change went in, but ooc, why do we overwrite the destinationChainId instead of just reverting if `destinationChainId != chainId()` ? If the caller has mis-specified that then I'd be concerned about other relayData attributes being wrong.

Reply:
hmm fair point

Reply:
Done 69db9ef7ecdf6f134c9126e0671580f2c8862b48